fhir-server/src/main/java/au/csiro/pathling/fhirpath/UntypedResourcePath.java

```diff
@@ -123,4 +124,23 @@ public UntypedResourcePath copy(@Nonnull final String expression,
         datasetWithColumns.getColumn(valueColumn), singular, thisColumn, typeColumn, possibleTypes);
   }
 
+  @Override
+  @Nonnull
+  public NonLiteralPath mergeWith(@Nonnull final FhirPath target,
+      @Nonnull final Dataset<Row> dataset, @Nonnull final String expression,
+      @Nonnull final Column idColumn, @Nonnull final Optional<Column> eidColumn,
+      @Nonnull final Column valueColumn, final boolean singular,
+      @Nonnull final Optional<Column> thisColumn) {
+    if (target instanceof UntypedResourcePath && possibleTypes
+        .equals(((UntypedResourcePath) target).possibleTypes)) {
+      // Two UntypedResourcePaths can be merged together if they have the same set of possible
+      // types.
```

Comment:
I think this could potentially be more lenient, that it the any combination of possibleTypes should be allowed by the results possibleTypes should be the union of both inputs.
Or per minimum it what should be allowed is merging when possibleTypes contain target.possibleTypes.

Reply:
Interesting... as far as I can tell, we never actually use `possibleTypes`. And I think this is actually ok, any resources which don't match just don't get added to the resulting collection.

So you can pass anything you like as an argument to `ofType` - as long as it is a valid resource type code. It is not actually checked against the `possibleTypes` of the input path.

The spec doesn't really say that you need to check, from my reading: http://hl7.org/fhirpath/#oftypetype-type-specifier-collection

So I actually think we could remove `possibleTypes` altogether... unless I am missing something.

Reply:
I've removed `possibleTypes` in e99e70a.